src/org/usfirst/frc4904/robot/RobotMap.java

```diff
@@ -57,55 +68,89 @@
 		public static Dump ballDumper;
 		public static BallInnie ballIntake;
 		public static Subsystem[] mainSubsystems;
 		public static CustomXbox driverXbox;
 		public static CustomJoystick operatorStick;
-		// Vision
 		public static AligningCamera alignCamera;
+		public static CustomPIDController chassisMC;
+		public static NavX navx;
+		public static SolenoidShifters shifter;
 	}
 	
 	public RobotMap() {
 		Component.pdp = new PDP();
-		// Chassis
+		Component.shifter = new SolenoidShifters(Port.Pneumatics.solenoidUp, Port.Pneumatics.solenoidDown);
+		Component.navx = new NavX(SerialPort.Port.kMXP);
+		Component.chassisMC = new CustomPIDController(0.01, 0.0, -0.02, RobotMap.Component.navx);
+		Component.chassisMC.setInputRange(-180, 180);
+		Component.chassisMC.setContinuous(true);
+		Component.leftWheelEncoder = new CANEncoder("LeftEncoder", Port.CAN.leftEncoder, false);
+		Component.rightWheelEncoder = new CANEncoder("RightEncoder", Port.CAN.rightEncoder, false);
+		Component.chassisEncoderMC = new CustomPIDController(0.001, 0.0, -0.002, new EncoderGroup(100, Component.leftWheelEncoder, Component.rightWheelEncoder));
+		Component.leftWheel = new Motor("LeftWheel", false, new AccelerationCap(Component.pdp), new CANTalon(Port.Motors.CAN.leftDriveA), new CANTalon(Port.Motors.CAN.leftDriveB));
+		Component.rightWheel = new Motor("RightWheel", false, new AccelerationCap(Component.pdp), new CANTalon(Port.Motors.CAN.rightDriveA), new CANTalon(Port.Motors.CAN.rightDriveB));
 		Component.leftWheelEncoder = new CANEncoder(Port.CAN.leftEncoder);
-		Component.leftWheelEncoder.setReverseDirection(true);
-		Component.leftWheel = new PositionEncodedMotor("leftWheel", new AccelerationCap(Component.pdp), new CustomPIDController(Component.leftWheelEncoder), new VictorSP(Port.PWM.leftDriveMotor));
-		Component.leftWheel.disablePID(); // TODO add encoders
-		// Component.leftWheel.setInverted(true);
 		Component.rightWheelEncoder = new CANEncoder(Port.CAN.rightEncoder);
-		Component.rightWheel = new PositionEncodedMotor("rightWheel", new AccelerationCap(Component.pdp), new CustomPIDController(Component.rightWheelEncoder), new VictorSP(Port.PWM.rightDriveMotor));
-		Component.rightWheel.disablePID(); // TODO add encoders
-		// Component.rightWheel.setInverted(false);
-		// Ball Intake
-		Component.ballIntake = new BallInnie(new VictorSP(Port.PWM.ballInnie));
-		// Ball Dumper
-		Component.ballDumper = new Dump(new VictorSP(Port.PWM.vomitElevator), new VictorSP(Port.PWM.vomitOuttakeRoller));
-		// Flywheel
-		Motor leftFlywheelMotor = new Motor(new VictorSP(Port.PWM.flywheelLeftMotor));
-		leftFlywheelMotor.setInverted(true);
-		Motor rightFlywheelMotor = new Motor(new VictorSP(Port.PWM.flywheelRightMotor));
-		rightFlywheelMotor.setInverted(false);
-		Component.flywheel = new Flywheel(leftFlywheelMotor, rightFlywheelMotor);
+		double inchesPerPulse = 2 * Math.PI * Constant.Chassis.WHEEL_RADIUS_INCHES / Constant.Chassis.WHEEL_PULSES_PER_REVOLUTION;
```

Comment:
This is a constant and should thus be used as a constant. Furthermore, you should be putting this where it is used as a constant in all caps, and not in RobotMap.